lib/rubocop/cop/rspec/repeated_description.rb

```diff
@@ -0,0 +1,82 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated description strings in example groups.
+      #
+      # @example
+      #
+      #     # bad
+      #     RSpec.describe User do
+      #       it 'is valid' do
+      #         # ...
+      #       end
+      #
+      #       it 'is valid' do
+      #         # ...
+      #       end
+      #     end
+      #
+      #     # good
+      #     RSpec.describe User do
+      #       it 'is valid when first and last name are present' do
+      #         # ...
+      #       end
+      #
+      #       it 'is valid when last name only is present' do
+      #         # ...
+      #       end
+      #     end
+      #
+      class RepeatedDescription < Cop
+        include RuboCop::Cop::ConfigurableEnforcedStyle
+
+        def_node_matcher :example, <<-PATTERN
+          (block $(send _ #{Examples::ALL.node_pattern_union} ...) ...)
+        PATTERN
+
+        # @!method scope_change?(node)
+        #
+        #   Detect if the node is an example group or shared example
+        #
+        #   Selectors which indicate that indicate we should stop searching
+        #
+        def_node_matcher :scope_change?,
+                         (ExampleGroups::ALL + SharedGroups::ALL).block_pattern
+
+        MSG = 'Repeated description'.freeze
+
+        def on_block(node)
+          return unless example_group?(node)
+
+          repeated_descriptions(node).each do |repeated_hook|
+            add_offense(repeated_hook, :expression)
+          end
+        end
+
+        private
+
+        def repeated_descriptions(node)
+          hooks_in_group(node)
+            .group_by(&:method_args)
+            .reject { |args, _| args.empty? }
+            .values
+            .reject(&:one?)
+            .flatten
+        end
+
+        def hooks_in_group(node, &block)
+          return to_enum(__method__, node) unless block_given?
+
+          node.each_child_node { |child| find_hooks(child, &block) }
+        end
+
+        def find_hooks(node, &block)
+          return if scope_change?(node)
+
+          example(node, &block)
```

Comment:
Sorry about that. I added some doc comments and I also renamed the misleading name... It was a holdover from code I borrowed from #254. Didn't realize I forgot to change the name sorry about that